### PR TITLE
Add `migrate-tables` workflow

### PR DESCRIPTION
**Pull request by qziyuan**
## Changes
Add migrate-tables workflow. This PR includes the workflow only described in milestone 1 in https://github.com/databrickslabs/ucx/issues/1035
table_migration_progress table is out of scope in this PR.

### Linked issues

Relates #333 #670

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [x] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [x] added unit tests
- [x] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: src/databricks/labs/ucx/install.py [hunk 1]
```diff
@@ -277,6 +277,27 @@ def _configure_new_installation(self) -> WorkspaceConfig:
 
         policy_id, instance_profile, spark_conf_dict = self._policy_installer.create(inventory_database)
 
+        # Save configurable spark_conf for table migration cluster
+        # parallelism will not be needed if backlog is fixed in https://databricks.atlassian.net/browse/ES-975874
+        parallelism = self._prompts.question(
+            "Parallelism for migrating dbfs root delta tables with deep clone", default="200", valid_number=True
+        )
+        if spark_conf_dict:
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:285]:
```suggestion
        if not spark_conf_dict: spark_conf_dict ={}
```

File: tests/unit/framework/test_tasks.py [hunk 2]
```diff
@@ -75,3 +82,40 @@ def _log_contents(tmp_path):
...
+
+def test_run_task(capsys):
+    # mock a task function to be tested
+    @task("migrate-tables", job_cluster="migration_sync")
+    def mock_migrate_external_tables_sync(cfg, workspace_client, sql_backend, installation):
+        """This mock task of migrate-tables"""
+        return f"Hello, World! {cfg} {workspace_client} {sql_backend} {installation}"
+
+    args = parse_args("--config=foo", "--task=mock_migrate_external_tables_sync", "--parent_run_id=abc", "--job_id=123")
+    cfg = create_autospec(WorkspaceConfig)
```
Comment (nfx) [inline tests/unit/framework/test_tasks.py:103]: It's a dataclass, init it

**Review by nfx — APPROVED** [review]
Yep, looks good. Add an integration test

File: src/databricks/labs/ucx/install.py (hunk 1 above)
Comment (qziyuan) [reply src/databricks/labs/ucx/install.py]: fixed

File: tests/unit/framework/test_tasks.py (hunk 2 above)
Comment (qziyuan) [reply tests/unit/framework/test_tasks.py]: fixed

**Review by nfx — APPROVED** [review]
Lgtm

**Pull request merged**